src/App.tsx

```diff
@@ -1,156 +1,175 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useContext, useEffect, useRef } from 'react';
+import { DispatchContext, StateContext } from './context/Store';
+import { Filter } from './types/Filter';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
+import { TodoItem } from './components/TodoItem';
+import { TodoFilter } from './components/TodoFilter';
 
 export const App: React.FC = () => {
+  const { todos, inputValue, filterStatus } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+
+  const inputElement = useRef<HTMLInputElement>(null);
+
+  const filteredTodos = todos.filter(todo => {
+    const { completed } = todo;
+
+    switch (filterStatus) {
+      case Filter.Active:
+        return !completed;
+
+      case Filter.Completed:
+        return completed;
+
+      default:
+        return todo;
+    }
+  });
```

Comment:
move filter inside switch case not vice versa. To avoid redundant iteration in default case